perfkitbenchmarker/windows_packages/ntttcp.py

```diff
@@ -0,0 +1,128 @@
+
+  shared_options = '-xml -t {time} -p {port} '.format(time=FLAGS.ntttcp_time,
+                                                      port=BASE_DATA_PORT)
+
+  client_options = '-s -m "{threads},*,{ip}"'.format(
+      threads=FLAGS.ntttcp_threads, ip=receiving_ip_address)
+  server_options = '-r -m "{threads},*,0.0.0.0"'.format(
+      threads=FLAGS.ntttcp_threads)
+
+  ntttcp_exe_dir = os.path.join(sending_vm.temp_dir, NTTTCP_DIR, 'x86')
+
+  # NTttcp will append to the xml file when it runs, which causes parsing
+  # to fail if there was a preexisting xml file. To be safe, try deleting
+  # the xml file.
+  rm_command = 'cd {ntttcp_exe_dir}; rm xml.txt'.format(
+      ntttcp_exe_dir=ntttcp_exe_dir)
+  sending_vm.RemoteCommand(rm_command, ignore_failure=True)
+
+  def _RunNtttcp(vm, options):
+    ntttcp_exe_dir = os.path.join(vm.temp_dir, NTTTCP_DIR, 'x86')
+    command = 'cd {ntttcp_exe_dir}; .\\NTttcp.exe {ntttcp_options}'.format(
+        ntttcp_exe_dir=ntttcp_exe_dir, ntttcp_options=options)
+    vm.RemoteCommand(command)
+
+  args = [((vm, shared_options + options), {}) for vm, options in
+          zip([sending_vm, receiving_vm], [client_options, server_options])]
+  vm_util.RunThreaded(_RunNtttcp, args)
+
+  cat_command = 'cd {ntttcp_exe_dir}; cat xml.txt'.format(
+      ntttcp_exe_dir=ntttcp_exe_dir)
+  stdout, _ = sending_vm.RemoteCommand(cat_command)
+
+  metadata = {
+      'receiving_machine_type': receiving_vm.machine_type,
+      'receiving_zone': receiving_vm.zone,
+      'sending_machine_type': sending_vm.machine_type,
+      'sending_zone': sending_vm.zone,
+      'ip_type': ip_type
+  }
+
+  return ParseNtttcpResults(stdout, metadata)
+
+
+def ParseNtttcpResults(xml_results, metadata):
+  """Parses the xml output from NTttcp and returns a list of samples.
+
+  The list of samples contains total throughput and per thread throughput
+  metrics (if there is more than a single thread).
+  """
+  root = xml.etree.ElementTree.fromstring(xml_results)
+  samples = []
+  metadata = metadata.copy()
+
+  for element in root.findall('parameters/*'):
+    if element.tag and element.text:
+      metadata[element.tag] = element.text
+
+  # TODO(ehankland): There is more interesting metadata that we can
+  # extract from the xml file such as the number of retransmits and
+  # cpu usage.
+
+  throughput_element = root.find('./throughput[@metric="mbps"]')
+  samples.append(sample.Sample(
+      'Total Throughput', float(throughput_element.text), 'mbps', metadata))
+
+  thread_elements = root.findall('./thread')
+  if len(thread_elements) > 1:
+    for element in thread_elements:
+      throughput_element = element.find('./throughput[@metric="mbps"]')
+      metadata = metadata.copy()
+      metadata['thread_index'] = element.attrib['index']
+      samples.append(sample.Sample('Thread Throughput',
+                                   float(throughput_element.text),
+                                   'mbps',
```

Comment:
Done - I was trying to be consistent with the tool, but being consistent with ourselves is probably better